src/_lib/modules/charts/tooltip/tooltip.directive.ts

```diff
@@ -0,0 +1,42 @@
+import {Directive, ElementRef, Input, OnChanges, OnInit, Renderer2, SimpleChanges} from '@angular/core';
+
+@Directive({
+    selector: '[nwChartTooltip]'
+})
+export class TooltipDirective implements OnChanges, OnInit {
+    @Input() position: [number, number];
+    @Input() chartWidth: number;
+    @Input() chartHeight: number;
+    @Input() chartMargins: {
+        top?: number;
+        bottom?: number;
+        left?: number;
+        right?: number;
+    };
+    @Input() offsetX: number = 16;
+    private _width: number;
+    private _height: number;
+
+    constructor(
+        private _elRef: ElementRef,
+        private _renderer: Renderer2) {
+    }
+
+    ngOnInit() {
+        this._width = this._elRef.nativeElement.getBoundingClientRect().width;
```

Comment:
Should only call `getBoundingClientRect()` once and use that to get the width and height

Reply:
@garethdn Isn't putting it inside ngOnInit() already constricting it to only being called once in the component's lifecycle?